### ISSUE TITLE
Support custom broker settings

### ISSUE THREAD

**Issue by Cs4r** [salesforce/kafka-junit#12]
KafkaTestServer doesn't support any way of overriding the kafka broker settings.

e.g: `num.io.threads`,  `auto.create.topics.enable`, `port` and `zookeeper.session.timeout.ms` are all hardcoded in `KafkaTestServer::start`.

**Comment by Crim** [comment]
Hi @Cs4r thanks for the suggestion!

I'd love to understand/talk through how exactly you'd like to see this work.  I could see a constructor argument being added to [KafkaTestServer](https://github.com/salesforce/kafka-junit/blob/master/kafka-junit-core/src/main/java/com/salesforce/kafka/test/KafkaTestServer.java#L90) that would allow for passing in broker settings.  That seems like a super easy thing to add.

I'm curious how you would see that being used by the [Kafka-JUnit4 ClassRule](https://github.com/salesforce/kafka-junit/blob/master/kafka-junit4/src/main/java/com/salesforce/kafka/test/junit4/SharedKafkaTestResource.java#L68-L70)?  Would you just pass it on the constructor for SharedKAfkaTestResource?

IE:

```java
@ClassRule
    public static final SharedKafkaTestResource sharedKafkaTestResource = new SharedKafkaTestResource(customBrokerProperties);
```

How about for the JUnit5 variant?  Perhaps something similar making use of the @RegisterExtension annotation instead of the @ExtendsWith annotation?

**Comment by Cs4r** [comment]
Hi @Crim,

Thanks for your quick response.

For me it would be very useful to tune some of the broker settings in order to proof that my custom consumer/producer behaves correctly in certain particular scenarios. e.g: what would happen if it tries to send a message to a topic that doesn't exist? That's not currently possible since `auto.create.topics.enable` is set to true. 

I also thought of adding an argument constructor to `KafkaTestServer `, such constructor would take a `java.util.Properties` containing only the properties you want to override. Your suggestions about the integration with JUnit4 and JUnit5 seen sensible to me.

It doesn't seem to be very complicated unless I am missing something.

Regards

**Comment by Crim** [comment]
@Cs4r Can you please review the READMEs to verify this new behavior will satisfy your needs?

[Kafka-JUnit4 README](https://github.com/salesforce/kafka-junit/tree/sp/ISSUE-12/kafka-junit4#kafkatestserver)

[Kafka-JUnit5 README](https://github.com/salesforce/kafka-junit/tree/sp/ISSUE-12/kafka-junit5#kafkatestserver)

**Comment by Cs4r** [comment]
Thank you for being so quick addressing the issue, I appreciate it.

**Comment by Crim** [comment]
I just published 2.3.0 to the maven repository.  If you run into any troubles or have any other great ideas feel free to submit a new issue.

Thanks!
